esp-box-lite/esp-box-lite.c

```diff
@@ -116,81 +125,20 @@ esp_err_t bsp_spiffs_unmount(void)
     return esp_vfs_spiffs_unregister(CONFIG_BSP_SPIFFS_PARTITION_LABEL);
 }
 
-
-esp_err_t bsp_audio_init(const i2s_std_config_t *i2s_config, i2s_chan_handle_t *tx_channel, i2s_chan_handle_t *rx_channel)
-{
-    if (i2s_tx_chan && i2s_rx_chan && i2s_data_if) {
-        if (tx_channel) {
-            *tx_channel = i2s_tx_chan;
-        }
-        if (rx_channel) {
-            *rx_channel = i2s_rx_chan;
-        }
-
-        /* Audio was initialized before */
-        return ESP_OK;
-    }
-
-    /* Setup I2S peripheral */
-    i2s_chan_config_t chan_cfg = I2S_CHANNEL_DEFAULT_CONFIG(CONFIG_BSP_I2S_NUM, I2S_ROLE_MASTER);
-    chan_cfg.auto_clear = true; // Auto clear the legacy data in the DMA buffer
-    BSP_ERROR_CHECK_RETURN_ERR(i2s_new_channel(&chan_cfg, tx_channel, rx_channel));
-
-    /* Setup I2S channels */
-    const i2s_std_config_t std_cfg_default = BSP_I2S_DUPLEX_MONO_CFG(22050);
-    const i2s_std_config_t *p_i2s_cfg = &std_cfg_default;
-    if (i2s_config != NULL) {
-        p_i2s_cfg = i2s_config;
-    }
-
-    if (tx_channel != NULL) {
-        BSP_ERROR_CHECK_RETURN_ERR(i2s_channel_init_std_mode(*tx_channel, p_i2s_cfg));
-        BSP_ERROR_CHECK_RETURN_ERR(i2s_channel_enable(*tx_channel));
-    }
-    if (rx_channel != NULL) {
-        BSP_ERROR_CHECK_RETURN_ERR(i2s_channel_init_std_mode(*rx_channel, p_i2s_cfg));
-        BSP_ERROR_CHECK_RETURN_ERR(i2s_channel_enable(*rx_channel));
-    }
-
-    audio_codec_i2s_cfg_t i2s_cfg = {
-        .port = CONFIG_BSP_I2S_NUM,
-        .rx_handle = i2s_rx_chan,
-        .tx_handle = i2s_tx_chan,
-    };
-    i2s_data_if = audio_codec_new_i2s_data(&i2s_cfg);
-    BSP_NULL_CHECK(i2s_data_if, NULL);
-
-    /* Setup power amplifier pin */
-    const gpio_config_t io_conf = {
-        .intr_type = GPIO_INTR_DISABLE,
-        .mode = GPIO_MODE_OUTPUT,
-        .pin_bit_mask = BIT64(BSP_POWER_AMP_IO),
-        .pull_down_en = GPIO_PULLDOWN_DISABLE,
-        .pull_up_en = GPIO_PULLDOWN_DISABLE,
-    };
-    BSP_ERROR_CHECK_RETURN_ERR(gpio_config(&io_conf));
-
-    return ESP_OK;
-}
-
-esp_err_t bsp_audio_poweramp_enable(bool enable)
-{
-    BSP_ERROR_CHECK_RETURN_ERR(gpio_set_level(BSP_POWER_AMP_IO, enable ? 1 : 0));
-
-    return ESP_OK;
-}
-
 esp_codec_dev_handle_t bsp_audio_codec_speaker_init(void)
 {
-    if (i2s_tx_chan == NULL || i2s_rx_chan == NULL || i2s_data_if == NULL) {
+    const audio_codec_data_if_t *i2s_data_if = bsp_audio_get_codec_itf();
+    if (i2s_data_if == NULL) {
         /* Initilize I2C */
         BSP_ERROR_CHECK_RETURN_ERR(bsp_i2c_init());
```

Comment:
Here we should use `BSP_ERROR_CHECK_RETURN_NULL()` to check the error. There are many same codes in `bsp_audio_codec_speaker_init()` and `bsp_audio_codec_microphone_init()` and other BSPs.

Reply:
Thank you very much for find this issue. I fixed it in ESP-BOX and ESP-BOX-Lite. For other boards will be fixed in next PRs.